### ISSUE TITLE
Object reference not set to an instance of an object

### ISSUE THREAD

**Issue by valkuc** [luandersonn/MyFTP#40]
Hi. I'm getting "Object reference not set to an instance of an object" when trying to connect to FTP. Could you please tell me where I can find application logs to provide you with more details?
![изображение](https://user-images.githubusercontent.com/6421082/194381243-4056ff3f-2025-4168-8464-d72dc58d71a1.png)

I found logs in settings but those elements are not clickable.
![изображение](https://user-images.githubusercontent.com/6421082/194381375-b6ea5ca2-af6e-470b-94c7-153c2520c45e.png)

**Comment by luandersonn** [comment]
Try click the icon

**Comment by valkuc** [comment]
Yep, that worked, but not intuitive - it looks like disabled. Anyway, I was able to get logs:
```
************************************************************** Logger started ***
[2022-10-06 20:49:41.174] >         ConnectAsync()
[2022-10-06 20:49:41.175] Status:   Connecting to ***:2121
[2022-10-06 20:49:41.197] Response: 220 ---------- Welcome to MiXplorer v6.44.0-Silver-B20030911 ----------
[2022-10-06 20:49:41.197] Command:  USER ***
[2022-10-06 20:49:41.205] Response: 331 USER send password
[2022-10-06 20:49:41.205] Command:  PASS ***
[2022-10-06 20:49:41.214] Response: 230 PASS access granted
[2022-10-06 20:49:41.214] Command:  FEAT
[2022-10-06 20:49:41.273] Response: 211-Features supported
Response: UTF8
Response: SIZE
Response: MDTM
Response: MFMT
Response: MLST Type*;Size*;Modify*;Perm
Response: REST STREAM
[2022-10-06 20:49:41.273] Response: 211 End
[2022-10-06 20:49:41.273] Status:   Text encoding: EETypeRva:0x0005E8A0
[2022-10-06 20:49:41.273] Command:  OPTS UTF8 ON
[2022-10-06 20:49:41.282] Response: 200 OPTS Done
[2022-10-06 20:49:41.282] Command:  SYST
[2022-10-06 20:49:41.289] Response: 215 UNIX Type: L8
[2022-10-06 20:49:41.290] Status:   Listing parser set to: Machine
[2022-10-06 20:49:41.308] >         GetObjectInfo("/", False)
[2022-10-06 20:49:41.308] Command:  MLST /
[2022-10-06 20:49:41.484] Response: 250 MLST /
[2022-10-06 20:49:41.627] System.NullReferenceException: Object reference not set to an instance of an object.
   at FluentFTP.FtpClient.<GetObjectInfoAsync>d__224.MoveNext() + 0x12c
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() + 0x21
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task) + 0x70
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task) + 0x38
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task) + 0x17
   at MyFTP.ViewModels.LoginViewModel.<LoginCommandAsync>d__33.MoveNext() + 0x4c2

************************************************************** Logger Dispose ***
```

**Comment by valkuc** [comment]
@luandersonn any idea?

**Comment by valkuc** [comment]
I have updated Mixplorer to the latest version and MyFTP is now able to connect without problems.

**Comment by luandersonn** [comment]
@valkuc Android FTP Server?

**Comment by valkuc** [comment]
@luandersonn not exactly, but yes, FTP server. The app name is Mixplorer and it's has ability to act as FTP, HTTP, SMB1 server

**Comment by luandersonn** [comment]
@valkuc Okay. Can I close this issue?

**Comment by valkuc** [comment]
@luandersonn yep, you can close it if you don't want to spent time on error handling, because it's not UX to display `Object reference is not set to an instance of an object` to a regular user :)

**Comment by codsec** [comment]
@valkuc did u find any other alternative like this software, because looks like the dev is just AFK for this project...

**Comment by valkuc** [comment]
> I have updated Mixplorer to the latest version and MyFTP is now able to connect without problems.

^^^

**Comment by codsec** [comment]
> > I have updated Mixplorer to the latest version and MyFTP is now able to connect without problems.
> 
> ^^^

could u tell me how can i do that exacly or share with me your version of this source or even the binary file(it's all i need actually)?

**Comment by valkuc** [comment]
Sorry, but if you faced that issue when using Mixplorer as a FTP server then you just need to update it on your smartphone. You can do that via Google Play or download it directly from official site. Just goooogle for it. But if you getting thus error with any other FTP server, then I can't help here.

**Comment by codsec** [comment]
> Sorry, but if you faced that issue when using Mixplorer as a FTP server then you just need to update it on your smartphone. You can do that via Google Play or download it directly from official site. Just goooogle for it. But if you getting thus error with any other FTP server, then I can't help here.

oh im not talking about the apk mixplorer, i mean, how did u get this source(MyFTP) working? because mine i just closes when i try to use's

**Comment by valkuc** [comment]
No, I did not do anything with MyFTP. For me the problem was solved just by updating FTP server app.

**Comment by codsec** [comment]
> No, I did not do anything with MyFTP. For me the problem was solved just by updating FTP server app.

could plz send me your release of the MyFTP? because for me when i try to compile nothing works

**Comment by valkuc** [comment]
> > No, I did not do anything with MyFTP. For me the problem was solved just by updating FTP server app.
> 
> could plz send me your release of the MyFTP? because for me when i try to compile nothing works

No, I can't send you binaries, I installed it from Microsoft Store.

**Comment by codsec** [comment]
> > > No, I did not do anything with MyFTP. For me the problem was solved just by updating FTP server app.
> > 
> > 
> > could plz send me your release of the MyFTP? because for me when i try to compile nothing works
> 
> No, I can't send you binaries, I installed it from Microsoft Store.

i try that one too, not working at all

**Comment by luandersonn** [comment]
I'm migrating the project to the Windows App SDK and hope to fix this bug in the new version #45

**Comment by luandersonn** [comment]
@valkuc @codsec 

> @valkuc did u find any other alternative like this software, because looks like the dev is just AFK for this project...

Sadly true. Balancing side projects and work is a difficult challenge. But I'm back :)

![image](https://github.com/luandersonn/MyFTP/assets/30560453/a9f59e06-fda4-4329-9121-44d8bb2e960f)

**Comment by codsec** [comment]
> I'm migrating the project to the Windows App SDK and hope to fix this bug in the new version #45

Could u make it standalone, without windows store etc?

**Comment by luandersonn** [comment]
> > I'm migrating the project to the Windows App SDK and hope to fix this bug in the new version #45
> 
> Could u make it standalone, without windows store etc?

Okay, I'll check the achievability
